File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        s = s.substring(2, s.length()-2);
        String[] text = s.split("},\\{");
        
        List<int[]> list = new ArrayList<>();
            
        for (String t : text) {
            int[] numbers = Arrays.stream(t.split(","))
                                  .mapToInt(Integer::parseInt)
                                  .toArray();
            list.add(numbers);
        }
        
        list.sort(Comparator.comparingInt(temp -> temp.length));
        
        List<Integer> answer = new ArrayList<>();
        
        for(int i=0; i<list.size(); i++) {
            for (int j=0; j<list.get(i).length; j++){
                if (!answer.contains(list.get(i)[j])) {
                    answer.add(list.get(i)[j]);
                }
            }
        }
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    
    s = s[2:-2].split('},{')
    li = sorted([list(map(int, i.split(','))) for i in s], key=len)
    
    for i in li:
        for j in i:
            if j not in answer:
                answer.append(j)
    
    return answer